### DISCUSSION TITLE
Control audio playback through an API

### DISCUSSION THREAD

**Discussion by vidikn** [librespot-org/librespot#1050] [Feature Requests]
Is it possible to control audio aspects through code without input from a Spotify client? (e.g. skip track, pause, etc)

**Comment by michaelherger** [comment]
Use the web API: https://developer.spotify.com/console/player/

**Comment by vidikn** [comment]
Does not really satisfy my use case since many clients would connect to that same speaker and I'd try to avoid getting OAuth API keys from each of them. Isn't there a way to do it locally, via librespot?

**Comment by anoFish** [comment]
I'm also looking for a way to do this. I want to run Librespot on a Raspberry Pi connected to a small display that shows the current player status and in addition I want to control the player with connected buttons.
If there is no such API on Librespot, maybe this would be an alternative: https://pypi.org/project/spotify-local-control/

**Comment by vidikn** [comment]
> I'm also looking for a way to do this. I want to run Librespot on a Raspberry Pi connected to a small display that shows the current player status and in addition I want to control the player with connected buttons.
> If there is no such API on Librespot, maybe this would be an alternative: https://pypi.org/project/spotify-local-control/

Thank you for the reply! This library only works if the Spotify app client is active on the PC right? Still wouldn't satisfy me if I had random phone users logging via zeroconf to the server...

**Comment by anoFish** [comment]
I haven't tried it yet, that's just what I found when I was looking for something like this I don't even know if this python project is still active. But I think there will be something along the lines that will cover this use case. Anyhow, an API at Librespot would definitely be useful and nice. Unfortunately I don't have any experience with Rust to implement it myself.

**Comment by eladyn** [comment]
There is [`spotifyd`](https://github.com/Spotifyd/spotifyd), which builds upon `librespot`. It offers, among other things, a DBus MPRIS interface, which can be used to query and modify the current playback state.

Do note however that:
- The project is not *that* active at the moment and the version of e.g. `librespot` that is used is still `0.2.0`.
- Depending on your exact use case, you may need some additions that haven't made it into a release yet. ([#1025](https://github.com/Spotifyd/spotifyd/pull/1025), [#954](https://github.com/Spotifyd/spotifyd/pull/954))

**Comment by roderickvd** [comment]
I moved this into a discussion, as it is not really an issue with `librespot` at the moment. Currently this is not in scope but I would consider PRs to get it in.
